networkx/algorithms/distance_measures.py

```diff
@@ -345,6 +345,12 @@ def diameter(G, e=None, usebounds=False, weight=None):
     e : eccentricity dictionary, optional
       A precomputed dictionary of eccentricities.
 
+    usebounds : bool, optional
+        If `True`, use the extrema bounding approach when computing the diameter
```

Comment:
It might be useful to include a reference to the [`_extrema_bounding`](https://github.com/networkx/networkx/blob/478183cb7c692a82dd529c58f20d05a17b271242/networkx/algorithms/distance_measures.py#L22) function here or add a line explaining the "extrema bounding approach". (same suggestion for the other docstrings as well). LMKWYT.

Reply:
The problem with this is it's not possible to link to the private `_extrema_bounding` function without adding the function to the docs so that the docstring stubs are generated. I'm personally -1 on adding the private function to the docs, as that subverts the motivation of making the function private in the first place.

If anything, I'd advocate for adding something to the notes sections of these docstrings.